### ISSUE TITLE
StringIndexOutOfBoundsException parsing list followed by unfenced code block

### ISSUE THREAD

**Issue by spearce** [atlassian/commonmark-java#52]
This Markdown fragment throws StringIndexOutOfBoundsException from DocumentParser:

``` java
    String markdown = "## Do this\n" +
        "- cd to foo\n" +
        "\n" +
        "\tgit clone https://...\n" +
        "\n" +
        "## Next\n";
```

```
java.lang.StringIndexOutOfBoundsException: String index out of range: 8
	at org.commonmark.internal.util.Substring.subSequence(Substring.java:50)
	at org.commonmark.internal.DocumentParser.addLine(DocumentParser.java:330)
	at org.commonmark.internal.DocumentParser.incorporateLine(DocumentParser.java:245)
	at org.commonmark.internal.DocumentParser.parse(DocumentParser.java:74)
	at org.commonmark.parser.Parser.parse(Parser.java:61)
```

I think the parser is confused exiting the unfenced code block and leaving columnIsInTab=true, resulting in `afterTab > line.length() ` and the parser exploding:

``` java
            // Our column is in a partially consumed tab. Expand the remaining columns (to the next tab stop) to spaces.
            int afterTab = index + 1;
            CharSequence rest = line.subSequence(afterTab, line.length());
```

**Comment by robinst** [comment]
Hi Shawn :)!

Thanks for reporting. Your analysis is right, the problem was that we weren't resetting it in `setNewIndex`. Pushed a fix and will release a new version shortly.

**Comment by robinst** [comment]
Released 0.5.1 with this fix: https://github.com/atlassian/commonmark-java/releases/tag/commonmark-parent-0.5.1
